MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -17,25 +17,17 @@ final class MapViewModel: ObservableObject {
     @Published var userPosition: MapCameraPosition = .userLocation(fallback: .automatic)
 
     init() {
-        readJsonFarmers()
+        getFarmers()
     }
 
     func onViewAppear() {
         requestUserAuthorization()
     }
 
-    private func readJsonFarmers() {
-        let decoder = JSONDecoder()
-        decoder.keyDecodingStrategy = .convertFromSnakeCase
-        do {
-            if let bundlePath = Bundle.main.path(forResource: "Farmers", ofType: "json"),
-               let farmerJson = try String(contentsOfFile: bundlePath).data(using: .utf8) {
-                let farmers = try decoder.decode([Farmer].self, from: farmerJson)
-                self.allFarmers = farmers
-            }
-        } catch {
-            print("Error decoding: \(error)")
-        }
+    private func getFarmers() {
+        let mapViewService = MapViewService()
```

Comment:
2 problems here:
- the code is not testable because you create the service inside the func. So if you want to test your viewModel you cannot provide a mock of the service
- If you need to refresh for instance, you will create the service each time you need it. 

One solution is to injecte the service in the view model init.